src/Workspaces/VisualBasic/Portable/Formatting/Rules/ElasticTriviaFormattingRule.vb

```diff
@@ -17,7 +17,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Formatting
         Public Sub New()
         End Sub
 
-        Public Overrides Sub AddSuppressOperations(list As List(Of SuppressOperation), node As SyntaxNode, lastToken As SyntaxToken, optionSet As OptionSet, nextOperation As NextAction(Of SuppressOperation))
+        Public Overrides Sub AddSuppressOperations(list As List(Of SuppressOperation), node As SyntaxNode, optionSet As OptionSet, nextOperation As NextAction(Of SuppressOperation))
```

Comment:
sorry, i've looked through the PR and i've realized there's something else i'm missing.  I can see the removal of "lastToken" from so many entyrpoints.  But i can't see the starting point where someone computed hte last token and passed it at the start so that it funneled through to all these locatoins.  Can you point me to that part of the PR?

Thanks!

Reply:
you added the last token, so probably it would be faster to find how you added it. I don't know why you added.

Reply:
Ok.  It seems like it's not a good to make such an invasive change then if we don't know well enough if this will be a problem.  I think more of an investigation has to be made here to fully understand the change and provide enough reasonable information to help reviewers assess if the new behavior is more desirable than the current behavior. 

> I don't know why you added.

Here are the comments i wrote:

```C#
            // We may think we have a complete set of braces, but that may not actually be the case
            // due incomplete code.  i.e. we have something like:
            //
            // class C
            // {
            //      int Blah {
            //          get { return blah
            // }
            //
            // In this case the parse will think that the get-accessor is actually on two lines 
            // (because it will consume the close curly that more accurately belongs to the class.
            //
            // Now there are different behaviors we want depending on what the user is doing 
            // and what we are formatting.  For example, if the user hits semicolon at the end of
            // "blah", then we want to keep the accessor on a single line.  In this scenario we
            // effectively want to ignore the following close curly as it may not be important to
            // this construct in the mind of the user. 
            //
            // However, say the user hits semicolon, then hits enter, then types a close curly.
            // In this scenario we would actually want the get-accessor to be formatted over multiple 
            // lines.  The difference here is that because the user just hit close-curly here we can 
            // consider it as being part of the closest construct and we can consider its placement
            // when deciding if the construct is on a single line.

                // In the case of a block, see if the first statement is on the same line 
                // as the open curly.  If so then we'll want to consider the end of the
                // block as the end of the first statement.  i.e. if you have:
                //
                //  try { }
                //  catch { return;     // <-- the end of this block is the end of the return statement.
                //  Method();
```

I've provided a large amount of commenting trying to give an intuition of what's going on here.  It would be helpful to have a similar level of depth provided to explain why it's ok that we don't perform this logic now in all cases adn that the change i behavior is acceptable and desirable.  Thanks!

Reply:
which is exactly the comment I linked when you were asking what it is doing.

Reply:
That comment describes the old behavior.  I'm tyring to understand how this works now that you've changed things to go through these new codepaths.  :)   I've asked that a few times now.  I still don't really understand the implication of this change.